src/routes/safe/components/Transactions/TxsTable/TxType/index.jsx

```diff
@@ -31,9 +31,29 @@ const typeToLabel = {
 }
 
 const TxType = ({ origin, txType }: { txType: TransactionType, origin: string | null }) => {
-  const iconUrl = txType === 'third-party-app' ? getAppInfo(origin).iconUrl : typeToIcon[txType]
-  const text = txType === 'third-party-app' ? getAppInfo(origin).name : typeToLabel[txType]
+  const isThirdPartyApp = txType === 'third-party-app'
+  const [loading, setLoading] = useState(true)
+  const [appInfo, setAppInfo] = useState()
 
-  return <IconText iconUrl={iconUrl} text={text} />
+  useEffect(() => {
+    const getAppInfo = async () => {
+      const parsedOrigin = getAppInfoFromOrigin(origin)
+      const appInfo = await getAppInfoFromUrl(parsedOrigin.url)
+      setAppInfo(appInfo)
+      setLoading(false)
+    }
+
+    if (!isThirdPartyApp) {
+      return
+    }
+
+    getAppInfo()
+  })
```

Comment:
no dependencies for the hook? is that intended?

Reply:
yes, but it has the same result as adding [txType], but I'll put it anyways, once we fix the never ending page refreshing it will work fine.

Reply:
done!